### DISCUSSION TITLE
SA 2.0 dataclasses: "hash" parameter from dataclasses.field missing in mapped_column()?

### DISCUSSION THREAD

**Discussion by arimeyer** [sqlalchemy/sqlalchemy#11894] [Usage Questions]
On standard dataclasses there is a **hash** parameter on dataclasses.field: https://docs.python.org/3/library/dataclasses.html#module-contents .  When trying to convert from SA 1.4 dataclasses to 2.0, I tried following:
https://docs.sqlalchemy.org/en/20/orm/dataclasses.html#attribute-configuration
https://docs.sqlalchemy.org/en/20/orm/mapping_api.html#sqlalchemy.orm.mapped_column

There was no mention about the "hash" parameter being specifically unsupported in mapped_column(), but when I add it to mirror my 1.4 code and set hash=False for certain fields, I get:
TypeError: __init__() got an unexpected keyword argument 'hash'

Was this deliberate?  If so, what is the workaround?

**Comment by CaselIT** [comment]
Hi,

Orm classes are mutable, so supporting hash that depends on the attributes values is usually not the best practice.

How come you need instances to be hashable on the attribute values?

As a workaround, you should be able to use 2.0 with the same mapping style you used in 1.4

**Comment by arimeyer** [comment]
Thanks for responding, Federico.  On these particular fields I want to ensure that they *won't* become part of the hash computation, so I mark them with _hash=False_.   I was marking most of my classes with _unsafe_hash=True_.  Is it preferable to leave that off to prevent them from being hashed at all (I assumed this was necessary for initial equality checks in Sets, Dicts, etc.).  I believe I was getting errors way back when I didn't have them marked with unsafe_hash.

In any case, was the _hash_ parameter left off mapped_column() deliberately?  If so, why keep unsafe_hash at the class level?

**Reply by CaselIT** [reply]
Oh, I was remembering that unsafe_hash was not available.
Since it is I think hash in the mapped_column was simply forgotten. If you want to send a pr it would be great!

Note that in most cases the default object hash that's derived from the `id()` value is enough in the orm, so I would not expect to need unsafe_has in most cases

**Reply by arimeyer** [reply]
Thanks, Federico.  I will try again, but for some reason I got errors when I didn't set `unsafe_hash=True`, which meant that if I didn't want a field to participate in the hash computation then I needed to set `hash=False` on that field.

I'll file an issue for this.

**Reply by CaselIT** [reply]
I've created #11923 to track it
